oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/cache/ServiceInstanceInventoryCache.java

```diff
@@ -119,4 +127,22 @@ public int getServiceInstanceId(int serviceId, int addressId) {
         }
         return serviceInstanceId;
     }
+
+    public String getServiceInstanceLanguage(int serviceInstanceId) {
+        String language = languageCache.getIfPresent(serviceInstanceId);
+        if (isNull(language)) {
+            ServiceInstanceInventory inventory = get(serviceInstanceId);
+            if (nonNull(inventory)) {
+                JsonObject properties = inventory.getProperties();
+                for (String key : properties.keySet()) {
+                    if (key.equals(ServiceInstanceInventory.PropertyUtil.LANGUAGE)) {
+                        language = properties.get(key).getAsString().toLowerCase();
+                        languageCache.put(serviceInstanceId, language);
+                        return language;
+                    }
+                }
+            }
+        }
+        return Const.EMPTY_STRING;
+    }
```

Comment:
Consider using "fast return", instead of so many nested blocks

Reply:
Done, thanks for your review :laughing: